petab/core.py

```diff
@@ -900,3 +900,87 @@ def measurements_have_replicates(measurement_df: pd.DataFrame) -> bool:
             measurement_df,
             ['observableId', 'simulationConditionId',
              'preequilibrationConditionId', 'time'])).size().values - 1)
+
+
+def flatten_timepoint_specific_output_overrides(
+        petab_problem: Problem) -> None:
+    """If the PEtab problem definition has timepoint-specific
+    observableParameters or noiseParameters for the same observable, replace
+    those by replicating the respective observable.
+
+    This is a helper function for some tools which may not support such
+    timepoint-specific mappings.
+
+    Arguments:
+        petab_problem:
+            PEtab problem to work on
+    """
+
+    df = petab_problem.measurement_df[
+        ["observableId",
+         "preequilibrationConditionId",
+         "simulationConditionId"]
+    ]
+    df_unique_values = df.drop_duplicates()
+    del df  # remove variable
+
+    # Create empty df -> to be filled with replicate-specific observables
+    df_new = pd.DataFrame()
+
+    for nrow in range(len(df_unique_values.index)):
+        df = petab_problem.measurement_df.loc[
+            (petab_problem.measurement_df['observableId'] >=
+             df_unique_values.loc[nrow, "observableId"])
+            & (petab_problem.measurement_df['preequilibrationConditionId'] <=
+               df_unique_values.loc[nrow, "preequilibrationConditionId"])
+            & (petab_problem.measurement_df['simulationConditionId'] <=
+               df_unique_values.loc[nrow, "simulationConditionId"])
+        ]
+
+        # get list of unique observable parameters
+        unique_sc = df["observableParameters"].unique()
+        # get list of unique noise parameters
+        unique_noise = df["noiseParameters"].unique()
+
+        for n_noise in range(len(unique_noise)):
+            for n_scale in range(len(unique_sc)):
+                idxs = (
+                        df["noiseParameters"].str.find(unique_noise[n_noise]) +
+                        df["observableParameters"].str.find(unique_sc[n_scale])
+                )
+                tmp = df.loc[idxs == 0, "observableId"]
+                df.loc[idxs == 0, "observableId"] = tmp+"_"+str(
+                    n_scale+n_noise+1
```

Comment:
flake8 will complain here :) always spaces around operators and between variables

Reply:
didn't complain :wink: but yes, I can change this